### ISSUE TITLE
pip-compile since 3.9.0 version failed when upgrade packages

### ISSUE THREAD

**Issue by krasnovmv** [jazzband/pip-tools#860]
Pip-compile failed when i try upgrade packages version.
I have local repositories in my requirements.in and some of them use other as install_requires. Before 3.9.0 all work fine, but after pip-compile try find local package in pip index, failed and show me error.

**Comment by atugushev** [comment]
Hello @krasnovmv,

Thanks for the report. Would be nice to have a steps to reproduce the bug. Could you at least attach full log of `pip-compile -v`, let's see how we can help you.

**Comment by krasnovmv** [comment]
> pip-compile requirements.in\prod.in --output-file requirements\prod.txt -v
Using indexes:
  http://pypi.cmp-dev/simple/

                          ROUND 1                           
Current constraints:
  celery (from -r requirements.in\main.in (line 25))
  Django==2.1.10 (from -r requirements.in\main.in (line 1))
  django-celery-beat (from -r requirements.in\main.in (line 26))
  django-extensions (from -r requirements.in\main.in (line 11))
  django-filter (from -r requirements.in\main.in (line 8))
  django-group-permissions from git+http://dev.companyname.local/packages/django-group-permissions.git@v1.1.13#egg=django-group-permissions (from -r requirements.in\prod.in (line 7))
  django-redis (from -r requirements.in\main.in (line 22))
  django-treebeard (from -r requirements.in\main.in (line 10))
  django-widget-tweaks (from -r requirements.in\main.in (line 9))
  django_xhtml2pdf (from -r requirements.in\main.in (line 40))
  djangorestframework (from -r requirements.in\main.in (line 4))
  djangorestframework-recursive (from -r requirements.in\main.in (line 5))
  drf-nested-routers (from -r requirements.in\main.in (line 6))
  gevent (from -r requirements.in\prod.in (line 3))
  greenlet (from -r requirements.in\prod.in (line 4))
  gunicorn (from -r requirements.in\prod.in (line 2))
  jsonschema (from -r requirements.in\main.in (line 32))
  cmp-django-dadata from git+http://dev.companyname.local/packages/cmp-django-dadata.git@v1.1.1#egg=cmp-django-dadata (from -r requirements.in\prod.in (line 8))
  cmp-django-es from git+http://dev.companyname.local/packages/cmp-django-es.git@v1.1.5#egg=cmp-django-es (from -r requirements.in\prod.in (line 9))
  cmp-django-gis from git+http://dev.companyname.local/cmp/cmp-django-gis.git@v0.0.9#egg=cmp-django-gis (from -r requirements.in\prod.in (line 14))
  cmp-django-morpher from git+http://dev.companyname.local/packages/cmp-django-morpher.git@v1.1.13#egg=cmp-django-morpher (from -r requirements.in\prod.in (line 10))
  cmp-django-utils from git+http://dev.companyname.local/packages/cmp-django-utils.git@v1.3.37#egg=cmp-django-utils (from -r requirements.in\prod.in (line 11))
  cmp-parseit-core from git+http://dev.companyname.local/parseit/cmp-parseit-core.git@v1.1.12#egg=cmp-parseit-core (from -r requirements.in\prod.in (line 12))
  cmp-parseit-drf from git+http://dev.companyname.local/parseit/cmp-parseit-drf.git@v1.0.39#egg=cmp-parseit-drf (from -r requirements.in\prod.in (line 13))
  openpyxl (from -r requirements.in\main.in (line 17))
  psycopg2 (from -r requirements.in\main.in (line 29))
  pyopenssl (from -r requirements.in\main.in (line 14))
  python-redis-lock (from -r requirements.in\main.in (line 20))
  raven (from -r requirements.in\prod.in (line 5))
  redis (from -r requirements.in\main.in (line 21))
  requests_file (from -r requirements.in\main.in (line 36))
  xmlschema (from -r requirements.in\main.in (line 37))
  zeep (from -r requirements.in\main.in (line 35))

Finding the best candidates:
  found candidate celery==4.3.0 (constraint was <any>)
  found candidate django==2.1.10 (constraint was ==2.1.10)
  found candidate django-celery-beat==1.5.0 (constraint was <any>)
  found candidate django-extensions==2.2.1 (constraint was <any>)
  found candidate django-filter==2.2.0 (constraint was <any>)
  found candidate git+http://dev.companyname.local/packages/django-group-permissions.git@v1.1.13#egg=django-group-permissions (constraint was <any>)
  found candidate django-redis==4.10.0 (constraint was <any>)
  found candidate django-treebeard==4.3 (constraint was <any>)
  found candidate django-widget-tweaks==1.4.5 (constraint was <any>)
  found candidate django-xhtml2pdf==0.0.4 (constraint was <any>)
  found candidate djangorestframework==3.10.1 (constraint was <any>)
  found candidate djangorestframework-recursive==0.1.2 (constraint was <any>)
  found candidate drf-nested-routers==0.91 (constraint was <any>)
  found candidate gevent==1.4.0 (constraint was <any>)
  found candidate greenlet==0.4.15 (constraint was <any>)
  found candidate gunicorn==19.9.0 (constraint was <any>)
  found candidate jsonschema==3.0.1 (constraint was <any>)
  found candidate git+http://dev.companyname.local/packages/cmp-django-dadata.git@v1.1.1#egg=cmp-django-dadata (constraint was <any>)
  found candidate git+http://dev.companyname.local/packages/cmp-django-es.git@v1.1.5#egg=cmp-django-es (constraint was <any>)
  found candidate git+http://dev.companyname.local/cmp/cmp-django-gis.git@v0.0.9#egg=cmp-django-gis (constraint was <any>)
  found candidate git+http://dev.companyname.local/packages/cmp-django-morpher.git@v1.1.13#egg=cmp-django-morpher (constraint was <any>)
  found candidate git+http://dev.companyname.local/packages/cmp-django-utils.git@v1.3.37#egg=cmp-django-utils (constraint was <any>)
  found candidate git+http://dev.companyname.local/parseit/cmp-parseit-core.git@v1.1.12#egg=cmp-parseit-core (constraint was <any>)
  found candidate git+http://dev.companyname.local/parseit/cmp-parseit-drf.git@v1.0.39#egg=cmp-parseit-drf (constraint was <any>)
  found candidate openpyxl==2.6.2 (constraint was <any>)
  found candidate psycopg2==2.8.3 (constraint was <any>)
  found candidate pyopenssl==19.0.0 (constraint was <any>)
  found candidate python-redis-lock==3.3.1 (constraint was <any>)
  found candidate raven==6.10.0 (constraint was <any>)
  found candidate redis==3.2.1 (constraint was <any>)
  found candidate requests-file==1.4.3 (constraint was <any>)
  found candidate xmlschema==1.0.13 (constraint was <any>)
  found candidate zeep==3.4.0 (constraint was <any>)

Finding secondary dependencies:
  djangorestframework==3.10.1 requires -
  pyopenssl==19.0.0         requires cryptography>=2.3, six>=1.5.2
  celery==4.3.0             requires billiard<4.0,>=3.6.0, kombu<5.0,>=4.4.0, pytz>dev, vine>=1.3.0
  django-widget-tweaks==1.4.5 requires -
  gevent==1.4.0             requires cffi>=1.11.5; sys_platform == "win32" and platform_python_implementation == "CPython", greenlet>=0.4.14; platform_python_implementation == "CPython"
  greenlet==0.4.15          requires -
  django-redis==4.10.0      requires Django>=1.11, redis>=2.10.0
  openpyxl==2.6.2           requires et_xmlfile, jdcal
  psycopg2==2.8.3           requires -
  redis==3.2.1              requires -
  raven==6.10.0             requires -
  zeep==3.4.0               requires appdirs>=1.4.0, attrs>=17.2.0, cached-property>=1.3.0, defusedxml>=0.4.1, isodate>=0.5.4, lxml>=3.1.0, pytz, requests-toolbelt>=0.7.1, requests>=2.7.0, six>=1.9.0
  xmlschema==1.0.13         requires elementpath~=1.1.7
  django-filter==2.2.0      requires Django>=1.11
  gunicorn==19.9.0          requires -
  jsonschema==3.0.1         requires attrs>=17.4.0, pyrsistent>=0.14.0, setuptools, six>=1.11.0
  django-xhtml2pdf==0.0.4   requires Django>=1.11, xhtml2pdf>=0.2.2
  djangorestframework-recursive==0.1.2 requires Django, djangorestframework>=3.0
  python-redis-lock==3.3.1  requires redis>=2.10.0
  requests-file==1.4.3      requires requests>=1.0.0, six
  django-extensions==2.2.1  requires six>=1.2
  django-celery-beat==1.5.0 requires django-timezone-field>=2.0, python-crontab>=2.3.4
  drf-nested-routers==0.91  requires Django>=1.11, djangorestframework>=3.6.0
  django==2.1.10            requires pytz
  django-treebeard==4.3     requires Django>=1.8

New dependencies found in this round:
  adding ['appdirs', '>=1.4.0', '[]']
  adding ['attrs', '>=17.2.0,>=17.4.0', '[]']
  adding ['billiard', '<4.0,>=3.6.0', '[]']
  adding ['cached-property', '>=1.3.0', '[]']
  adding ['celery', '>=4.3.0', '[]']
  adding ['cffi', '>=1.11.5', '[]']
  adding ['contextlib2', '==0.5.4', '[]']
  adding ['cryptography', '>=2.3', '[]']
  adding ['defusedxml', '>=0.4.1', '[]']
  adding ['django', '>=1.10.1,>=1.11,>=1.8', '[]']
  adding ['django-timezone-field', '>=2.0', '[]']
  adding ['djangorestframework', '>=3.0,>=3.6.0', '[]']
  adding ['djangorestframework-recursive', '', '[]']
  adding ['elementpath', '~=1.1.7', '[]']
  adding ['et-xmlfile', '', '[]']
  adding ['greenlet', '>=0.4.14', '[]']
  adding ['isodate', '>=0.5.4', '[]']
  adding ['jdcal', '', '[]']
  adding ['kombu', '<5.0,>=4.4.0', '[]']
  adding ['cmp-django-morpher', '', '[]']
  adding ['cmp-django-utils', '', '[]']
  adding ['cmp-parseit-core', '', '[]']
  adding ['lxml', '>=3.1.0', '[]']
  adding ['orderedset', '', '[]']
  adding ['psutil', '', '[]']
  adding ['pyjwt', '', '[]']
  adding ['pyrsistent', '>=0.14.0', '[]']
  adding ['python-crontab', '>=2.3.4', '[]']
  adding ['pytz', '>dev', '[]']
  adding ['redis', '>=2.10.0', '[]']
  adding ['requests', '>=1.0.0,>=2.7.0', '[]']
  adding ['requests-toolbelt', '>=0.7.1', '[]']
  adding ['setuptools', '', '[]']
  adding ['six', '>=1.11.0,>=1.2,>=1.5.2,>=1.9.0', '[]']
  adding ['vine', '>=1.3.0', '[]']
  adding ['xhtml2pdf', '>=0.2.2', '[]']
Removed dependencies in this round:
------------------------------------------------------------
Result of round 1: not stable

                          ROUND 2                           
Current constraints:
  appdirs>=1.4.0 (from zeep==3.4.0->-r requirements.in\main.in (line 35))
  attrs>=17.2.0,>=17.4.0 (from zeep==3.4.0->-r requirements.in\main.in (line 35))
  billiard<4.0,>=3.6.0 (from celery==4.3.0->-r requirements.in\main.in (line 25))
  cached-property>=1.3.0 (from zeep==3.4.0->-r requirements.in\main.in (line 35))
  celery>=4.3.0 (from -r requirements.in\main.in (line 25))
  cffi>=1.11.5 (from gevent==1.4.0->-r requirements.in\prod.in (line 3))
  contextlib2==0.5.4 (from cmp-django-utils->-r requirements.in\prod.in (line 11))
  cryptography>=2.3 (from pyopenssl==19.0.0->-r requirements.in\main.in (line 14))
  defusedxml>=0.4.1 (from zeep==3.4.0->-r requirements.in\main.in (line 35))
  Django==2.1.10,>=1.10.1,>=1.11,>=1.8 (from -r requirements.in\main.in (line 1))
  django-celery-beat (from -r requirements.in\main.in (line 26))
  django-extensions (from -r requirements.in\main.in (line 11))
  django-filter (from -r requirements.in\main.in (line 8))
  django-group-permissions from git+http://dev.companyname.local/packages/django-group-permissions.git@v1.1.13#egg=django-group-permissions (from -r requirements.in\prod.in (line 7))
  django-redis (from -r requirements.in\main.in (line 22))
  django-timezone-field>=2.0 (from django-celery-beat==1.5.0->-r requirements.in\main.in (line 26))
  django-treebeard (from -r requirements.in\main.in (line 10))
  django-widget-tweaks (from -r requirements.in\main.in (line 9))
  django_xhtml2pdf (from -r requirements.in\main.in (line 40))
  djangorestframework>=3.0,>=3.6.0 (from -r requirements.in\main.in (line 4))
  djangorestframework-recursive (from -r requirements.in\main.in (line 5))
  drf-nested-routers (from -r requirements.in\main.in (line 6))
  elementpath~=1.1.7 (from xmlschema==1.0.13->-r requirements.in\main.in (line 37))
  et_xmlfile (from openpyxl==2.6.2->-r requirements.in\main.in (line 17))
  gevent (from -r requirements.in\prod.in (line 3))
  greenlet>=0.4.14 (from -r requirements.in\prod.in (line 4))
  gunicorn (from -r requirements.in\prod.in (line 2))
  isodate>=0.5.4 (from zeep==3.4.0->-r requirements.in\main.in (line 35))
  jdcal (from openpyxl==2.6.2->-r requirements.in\main.in (line 17))
  jsonschema (from -r requirements.in\main.in (line 32))
  kombu<5.0,>=4.4.0 (from celery==4.3.0->-r requirements.in\main.in (line 25))
  cmp-django-dadata from git+http://dev.companyname.local/packages/cmp-django-dadata.git@v1.1.1#egg=cmp-django-dadata (from -r requirements.in\prod.in (line 8))
  cmp-django-es from git+http://dev.companyname.local/packages/cmp-django-es.git@v1.1.5#egg=cmp-django-es (from -r requirements.in\prod.in (line 9))
  cmp-django-gis from git+http://dev.companyname.local/cmp/cmp-django-gis.git@v0.0.9#egg=cmp-django-gis (from -r requirements.in\prod.in (line 14))
  cmp-django-morpher (from -r requirements.in\prod.in (line 10))
  cmp-django-utils (from -r requirements.in\prod.in (line 11))
  cmp-parseit-core (from -r requirements.in\prod.in (line 12))
  cmp-parseit-drf from git+http://dev.companyname.local/parseit/cmp-parseit-drf.git@v1.0.39#egg=cmp-parseit-drf (from -r requirements.in\prod.in (line 13))
  lxml>=3.1.0 (from zeep==3.4.0->-r requirements.in\main.in (line 35))
  openpyxl (from -r requirements.in\main.in (line 17))
  orderedset (from cmp-django-utils->-r requirements.in\prod.in (line 11))
  psutil (from cmp-django-utils->-r requirements.in\prod.in (line 11))
  psycopg2 (from -r requirements.in\main.in (line 29))
  PyJWT (from cmp-django-es->-r requirements.in\prod.in (line 9))
  pyopenssl (from -r requirements.in\main.in (line 14))
  pyrsistent>=0.14.0 (from jsonschema==3.0.1->-r requirements.in\main.in (line 32))
  python-crontab>=2.3.4 (from django-celery-beat==1.5.0->-r requirements.in\main.in (line 26))
  python-redis-lock (from -r requirements.in\main.in (line 20))
  pytz>dev (from zeep==3.4.0->-r requirements.in\main.in (line 35))
  raven (from -r requirements.in\prod.in (line 5))
  redis>=2.10.0 (from -r requirements.in\main.in (line 21))
  requests>=1.0.0,>=2.7.0 (from zeep==3.4.0->-r requirements.in\main.in (line 35))
  requests_file (from -r requirements.in\main.in (line 36))
  requests-toolbelt>=0.7.1 (from zeep==3.4.0->-r requirements.in\main.in (line 35))
  setuptools (from jsonschema==3.0.1->-r requirements.in\main.in (line 32))
  six>=1.11.0,>=1.2,>=1.5.2,>=1.9.0 (from zeep==3.4.0->-r requirements.in\main.in (line 35))
  vine>=1.3.0 (from celery==4.3.0->-r requirements.in\main.in (line 25))
  xhtml2pdf>=0.2.2 (from django-xhtml2pdf==0.0.4->-r requirements.in\main.in (line 40))
  xmlschema (from -r requirements.in\main.in (line 37))
  zeep (from -r requirements.in\main.in (line 35))

Finding the best candidates:
  found candidate appdirs==1.4.3 (constraint was >=1.4.0)
  found candidate attrs==19.1.0 (constraint was >=17.2.0,>=17.4.0)
  found candidate billiard==3.6.0.0 (constraint was >=3.6.0,<4.0)
  found candidate cached-property==1.5.1 (constraint was >=1.3.0)
  found candidate celery==4.3.0 (constraint was >=4.3.0)
  found candidate cffi==1.12.3 (constraint was >=1.11.5)
  found candidate contextlib2==0.5.4 (constraint was ==0.5.4)
  found candidate cryptography==2.7 (constraint was >=2.3)
  found candidate defusedxml==0.6.0 (constraint was >=0.4.1)
  found candidate django==2.1.10 (constraint was >=1.10.1,>=1.11,>=1.8,==2.1.10)
  found candidate django-celery-beat==1.5.0 (constraint was <any>)
  found candidate django-extensions==2.2.1 (constraint was <any>)
  found candidate django-filter==2.2.0 (constraint was <any>)
  found candidate git+http://dev.companyname.local/packages/django-group-permissions.git@v1.1.13#egg=django-group-permissions (constraint was <any>)
  found candidate django-redis==4.10.0 (constraint was <any>)
  found candidate django-timezone-field==3.0 (constraint was >=2.0)
  found candidate django-treebeard==4.3 (constraint was <any>)
  found candidate django-widget-tweaks==1.4.5 (constraint was <any>)
  found candidate django-xhtml2pdf==0.0.4 (constraint was <any>)
  found candidate djangorestframework==3.10.1 (constraint was >=3.0,>=3.6.0)
  found candidate djangorestframework-recursive==0.1.2 (constraint was <any>)
  found candidate drf-nested-routers==0.91 (constraint was <any>)
  found candidate elementpath==1.1.8 (constraint was ~=1.1.7)
  found candidate et-xmlfile==1.0.1 (constraint was <any>)
  found candidate gevent==1.4.0 (constraint was <any>)
  found candidate greenlet==0.4.15 (constraint was >=0.4.14)
  found candidate gunicorn==19.9.0 (constraint was <any>)
  found candidate isodate==0.6.0 (constraint was >=0.5.4)
  found candidate jdcal==1.4.1 (constraint was <any>)
  found candidate jsonschema==3.0.1 (constraint was <any>)
  found candidate kombu==4.6.3 (constraint was >=4.4.0,<5.0)
  found candidate git+http://dev.companyname.local/packages/cmp-django-dadata.git@v1.1.1#egg=cmp-django-dadata (constraint was <any>)
  found candidate git+http://dev.companyname.local/packages/cmp-django-es.git@v1.1.5#egg=cmp-django-es (constraint was <any>)
  found candidate git+http://dev.companyname.local/cmp/cmp-django-gis.git@v0.0.9#egg=cmp-django-gis (constraint was <any>)
Could not find a version that matches cmp-django-morpher (from -r requirements.in\prod.in (line 10))
No versions found
Was http://pypi.cmp-dev/simple/ reachable?


How i can see, pip-tools cant find repo for cmp-django-morher even it added from repo.

**Comment by atugushev** [comment]
@krasnovmv 

Have you tried earlier versions of `pip-tools`?

**Comment by atugushev** [comment]
FTR:  regarding the log:

```
                ROUND 1

Current constraints:
  ...
  cmp-django-morpher from git+http://dev.companyname.local/packages/cmp-django-morpher.git@v1.1.13#egg=cmp-django-morpher (from -r requirements.in\prod.in (line 10))
  ...

                ROUND 2

Current constraints:
  ...
  cmp-django-morpher (from -r requirements.in\prod.in (line 10))
  ...
```

the `cmp-django-morpher` has lost git repo URL on the second round.

**Comment by krasnovmv** [comment]
> @krasnovmv
> 
> Have you tried earlier versions of `pip-tools`? Which one is worked for you?

All is working on pip-tools 3.8.0

**Comment by atugushev** [comment]
Hi @krasnovmv. Could you please check whether the bug reproduces against current master?

**Comment by krasnovmv** [comment]
@atugushev, wow, i check and all work correctly. 
Thanks for resolve this issue, im going wait next npm release.

**Comment by atugushev** [comment]
Thanks to @andersk. Fixed in #879.

**Comment by atugushev** [comment]
`pip-tools v4.1.0` is released.
